### PR TITLE
Use reference name instead of field name in Query expression builder

### PR DESCRIPTION
**Pull request by ossinkine**

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Query/Expr.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function references($document)
                 }
 
                 foreach ($keys as $key => $value) {
-                    $this->query[$this->currentField . '.$' . $key] = $dbRef['$' . $key];
+                    $this->query[$mapping['name'] . '.$' . $key] = $dbRef['$' . $key];
```
Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Query/Expr.php]: Sorry for long time that has passed, but I think this also should be fixed in `includesReferenceTo` method below, would you mind?

Comment (ossinkine) [inline lib/Doctrine/ODM/MongoDB/Query/Expr.php]: I'm not sure how this method should work properly and it's not covered by tests.
So I made this changes on your own risk.

Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Query/Expr.php]: I'll take the responsibility :)

**Comment by malarzm** [comment]
Merged manually in https://github.com/doctrine/mongodb-odm/commit/0d4c2f06f3f02583af3304e241cb5b1319992d0c, thanks @ossinkine!

**Pull request closed without merging**